File: 2026-08-09-java-http-server/tests/TestRunner.java
public class TestRunner {
    public static void main(String[] args) throws Exception {
        HttpRequestParserTests.run();
        RouterTests.run();
        StaticFileHandlerTests.run();
        EndToEndTests.run();

        System.out.println();
        System.out.println(TestHelper.passed + " passed, " + TestHelper.failed + " failed");
        if (TestHelper.failed > 0) System.exit(1);
    }
}

File: 2026-08-14-fsharp-spreadsheet/tests/TestRunner.fs
module Spreadsheet.Tests.TestRunner

open Spreadsheet.Tests

[<EntryPoint>]
let main _ =
    ColumnCodecTests.run ()
    LexerTests.run ()
    ParserTests.run ()
    EvalTests.run ()
    GraphTests.run ()
    SheetTests.run ()

    printfn "\n%d passed, %d failed" TestHelper.Passed TestHelper.Failed
    if TestHelper.Failed > 0 then 1 else 0
